src/main/java/nextstep/courses/domain/registration/SessionRegistration.java

```diff
@@ -0,0 +1,47 @@
+package nextstep.courses.domain.registration;
+
+import nextstep.users.domain.NsUser;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SessionRegistration {
+    private final SessionStatus sessionStatus;
+    private final Students students;
+
+    public SessionRegistration(SessionStatus sessionStatus, int maxUserCount) {
+        this(sessionStatus, maxUserCount, new HashSet<>());
+    }
+
+    public SessionRegistration(SessionStatus sessionStatus, int maxUserCount, Set<NsUser> users) {
+        this(sessionStatus, new Students(maxUserCount, users));
+    }
+
+    public SessionRegistration(SessionStatus sessionStatus, Students students) {
+        this.sessionStatus = sessionStatus;
+        this.students = students;
+    }
+
+    public void addUser(NsUser user) {
```

Comment:
```suggestion
    public Student addUser(NsUser user) {
```

수강 신청이 가능할 경우 Session과 NsUser 사이의 관계가 만들어 지면서 Student(sessionId, nsUserid)와 같은 객체가 생성되지 않을까?